Dependency resolving error with custom maven repo, multiple runs and a "--deps" declaration

**Describe the bug**
Calling jbang is failing.

The context:

* We run `./jbang info tools`
* We have a custom company maven repo
* We use `RELEASE` as version definition (to get the latest release)

I think the issue depends of the state of the local maven repo (working on this to be sure)

**To Reproduce**

The script `myJangScript` has an custom company repo (which aggregates a maven-central mirror and a company maven repo).
Something like:

```
//REPOS acme=https://maven.acme.local/maven
```

We would like to use the latest version of `company-lib`

Therefore we are passing `--deps company.group.id:company-lib:RELEASE`.

In some situations (that I am trying to reproduce now), we have this error:

```
+ ./jbang info tools --verbose --deps company.group.id:company-lib:RELEASE myJangScript
 Picked up JAVA_TOOL_OPTIONS: -Dfile.encoding=UTF-8
 [jbang] jbang version 0.91.0
 [jbang] System Java version detected as 8
 [jbang] System Java version matches requested version 8
 [jbang] Resolving dependencies...
 [jbang] company.group.id:company-lib:jar:RELEASE
 [jbang] [ERROR] Could not resolve dependencies from mavencentral=https://repo1.maven.org/maven2/
 Unable to collect/resolve dependency tree for a resolution due to: Failed to collect dependencies at company.group.id:company-lib:jar:RELEASE, caused by: Could not find metadata company.group.id:company-lib/maven-metadata.xml in local (/var/lib/jenkins/.m2/repository)
   Failed to collect dependencies at company.group.id:company-lib:jar:RELEASE
 dev.jbang.cli.ExitException: Could not resolve dependencies from mavencentral=https://repo1.maven.org/maven2/
 Unable to collect/resolve dependency tree for a resolution due to: Failed to collect dependencies at company.group.id:company-lib:jar:RELEASE, caused by: Could not find metadata company.group.id:company-lib/maven-metadata.xml in local (/var/lib/jenkins/.m2/repository)
   Failed to collect dependencies at company.group.id:company-lib:jar:RELEASE
 	at dev.jbang.dependencies.DependencyUtil.resolveDependenciesViaAether(DependencyUtil.java:254)
 	at dev.jbang.dependencies.DependencyUtil.resolveDependencies(DependencyUtil.java:122)
 	at dev.jbang.dependencies.DependencyUtil.resolveDependencies(DependencyUtil.java:71)
 	at dev.jbang.dependencies.DependencyResolver.resolve(DependencyResolver.java:67)
 	at dev.jbang.source.RunContext.resolveClassPath(RunContext.java:354)
 	at dev.jbang.cli.BaseInfoCommand$ScriptInfo.<init>(Info.java:95)
 	at dev.jbang.cli.BaseInfoCommand.getInfo(Info.java:163)
 	at dev.jbang.cli.Tools.doCall(Info.java:186)
 	at dev.jbang.cli.BaseCommand.call(BaseCommand.java:93)
 	at dev.jbang.cli.BaseCommand.call(BaseCommand.java:15)
 	at picocli.CommandLine.executeUserObject(CommandLine.java:1953)
 	at picocli.CommandLine.access$1300(CommandLine.java:145)
 	at picocli.CommandLine$RunLast.executeUserObjectOfLastSubcommandWithSameParent(CommandLine.java:2352)
 	at picocli.CommandLine$RunLast.handle(CommandLine.java:2346)
 	at dev.jbang.cli.JBang$3.handle(JBang.java:145)
 	at dev.jbang.cli.JBang$3.handle(JBang.java:140)
 	at picocli.CommandLine$AbstractParseResultHandler.execute(CommandLine.java:2179)
 	at picocli.CommandLine.execute(CommandLine.java:2078)
 	at dev.jbang.Main.main(Main.java:14)
 Caused by: org.jboss.shrinkwrap.resolver.api.NoResolvedResultException: Unable to collect/resolve dependency tree for a resolution due to: Failed to collect dependencies at company.group.id:company-lib:jar:RELEASE, caused by: Could not find metadata company.group.id:company-lib/maven-metadata.xml in local (/var/lib/jenkins/.m2/repository)
 	at org.jboss.shrinkwrap.resolver.impl.maven.MavenWorkingSessionImpl.wrapException(MavenWorkingSessionImpl.java:503)
 	at org.jboss.shrinkwrap.resolver.impl.maven.MavenWorkingSessionImpl.resolveDependencies(MavenWorkingSessionImpl.java:242)
 	at org.jboss.shrinkwrap.resolver.impl.maven.MavenStrategyStageBaseImpl.using(MavenStrategyStageBaseImpl.java:70)
 	at org.jboss.shrinkwrap.resolver.impl.maven.MavenStrategyStageBaseImpl.withTransitivity(MavenStrategyStageBaseImpl.java:52)
 	at org.jboss.shrinkwrap.resolver.impl.maven.MavenStrategyStageBaseImpl.withTransitivity(MavenStrategyStageBaseImpl.java:39)
 	at dev.jbang.dependencies.DependencyUtil.resolveDependenciesViaAether(DependencyUtil.java:232)
 	... 18 more
 Caused by: org.eclipse.aether.resolution.DependencyResolutionException: Failed to collect dependencies at company.group.id:company-lib:jar:RELEASE
 	at org.eclipse.aether.internal.impl.DefaultRepositorySystem.resolveDependencies(DefaultRepositorySystem.java:353)
 	at org.jboss.shrinkwrap.resolver.impl.maven.bootstrap.MavenRepositorySystem.resolveDependencies(MavenRepositorySystem.java:121)
 	at org.jboss.shrinkwrap.resolver.impl.maven.MavenWorkingSessionImpl.resolveDependencies(MavenWorkingSessionImpl.java:239)
 	... 22 more
 Caused by: org.eclipse.aether.collection.DependencyCollectionException: Failed to collect dependencies at company.group.id:company-lib:jar:RELEASE
 	at org.eclipse.aether.internal.impl.collect.DefaultDependencyCollector.collectDependencies(DefaultDependencyCollector.java:291)
 	at org.eclipse.aether.internal.impl.DefaultRepositorySystem.resolveDependencies(DefaultRepositorySystem.java:309)
 	... 24 more
 Caused by: org.eclipse.aether.resolution.ArtifactDescriptorException: Failed to read artifact descriptor for company.group.id:company-lib:jar:RELEASE
 	at org.apache.maven.repository.internal.DefaultArtifactDescriptorReader.loadPom(DefaultArtifactDescriptorReader.java:218)
 	at org.apache.maven.repository.internal.DefaultArtifactDescriptorReader.readArtifactDescriptor(DefaultArtifactDescriptorReader.java:171)
 	at org.eclipse.aether.internal.impl.collect.DefaultDependencyCollector.resolveCachedArtifactDescriptor(DefaultDependencyCollector.java:541)
 	at org.eclipse.aether.internal.impl.collect.DefaultDependencyCollector.getArtifactDescriptorResult(DefaultDependencyCollector.java:524)
 	at org.eclipse.aether.internal.impl.collect.DefaultDependencyCollector.processDependency(DefaultDependencyCollector.java:412)
 	at org.eclipse.aether.internal.impl.collect.DefaultDependencyCollector.processDependency(DefaultDependencyCollector.java:365)
 	at org.eclipse.aether.internal.impl.collect.DefaultDependencyCollector.process(DefaultDependencyCollector.java:352)
 	at org.eclipse.aether.internal.impl.collect.DefaultDependencyCollector.collectDependencies(DefaultDependencyCollector.java:254)
 	... 25 more
 Caused by: org.eclipse.aether.resolution.VersionResolutionException: Failed to resolve version for company.group.id:company-lib:jar:RELEASE: Could not find metadata company.group.id:company-lib/maven-metadata.xml in local (/var/lib/jenkins/.m2/repository)
 	at org.apache.maven.repository.internal.DefaultVersionResolver.resolveVersion(DefaultVersionResolver.java:276)
 	at org.apache.maven.repository.internal.DefaultArtifactDescriptorReader.loadPom(DefaultArtifactDescriptorReader.java:204)
 	... 32 more
 Caused by: org.eclipse.aether.transfer.MetadataNotFoundException: Could not find metadata company.group.id:company-lib/maven-metadata.xml in local (/var/lib/jenkins/.m2/repository)
 	at org.eclipse.aether.internal.impl.DefaultMetadataResolver.resolve(DefaultMetadataResolver.java:220)
 	at org.eclipse.aether.internal.impl.DefaultMetadataResolver.resolveMetadata(DefaultMetadataResolver.java:181)
 	at org.apache.maven.repository.internal.DefaultVersionResolver.resolveVersion(DefaultVersionResolver.java:213)
 	... 33 more
 [jbang] If you believe this a bug in jbang, open an issue at https://github.com/jbangdev/jbang/issues
```

**Expected behavior**
The call works, I will post a example where it works (other Jenkins worker where the local maven repo is not the same)

**JBang version**
0.91.0

**Additional context**

First wired thing, this looks really wrong:

`[ERROR] Could not resolve dependencies from mavencentral=https://repo1.maven.org/maven2/`

For me jbang should use the repo defined in the source.

Is it different if I run `./jbang run` vs `./jbang info tools`

In my CI script I am running:

```
./jbang info tools --verbose --deps company.group.id:company-lib:RELEASE myJangScript
./jbang --verbose --deps company.group.id:company-lib:RELEASE myJangScript foo bar
```

And on a run with a clean local maven repo I now got:

* First run OK.
* Second run is wired:

```
./jbang --verbose --deps company.group.id:company-lib:RELEASE myJangScript foo bar
 Picked up JAVA_TOOL_OPTIONS: -Dfile.encoding=UTF-8
 [jbang] jbang version 0.91.0
 [jbang] Building as previous build jar found but it or its dependencies not up-to-date.
 [jbang] System Java version detected as 8
 [jbang] System Java version matches requested version 8
 [jbang] Building jar...
 [jbang] compile: javac -classpath /var/lib/jenkins/.m2/repository/info/picocli/picocli/4.6.3/picocli-4.6.3.jar:/var/lib/jenkins/.m2/repository/org/junit/platform/junit-platform-console/1.8.2/junit-platform-console-1.8.2.jar:/var/lib/jenkins/.m2/repository/org/junit/platform/junit-platform-reporting/1.8.2/junit-platform-reporting-1.8.2.jar:/var/lib/jenkins/.m2/repository/org/junit/platform/junit-platform-launcher/1.8.2/junit-platform-launcher-1.8.2.jar:/var/lib/jenkins/.m2/repository/org/apiguardian/apiguardian-api/1.1.2/apiguardian-api-1.1.2.jar:/var/lib/jenkins/.m2/repository/org/junit/jupiter/junit-jupiter-engine/5.8.2/junit-jupiter-engine-5.8.2.jar:/var/lib/jenkins/.m2/repository/org/junit/platform/junit-platform-engine/1.8.2/junit-platform-engine-1.8.2.jar:/var/lib/jenkins/.m2/repository/org/opentest4j/opentest4j/1.2.0/opentest4j-1.2.0.jar:/var/lib/jenkins/.m2/repository/org/junit/platform/junit-platform-commons/1.8.2/junit-platform-commons-1.8.2.jar:/var/lib/jenkins/.m2/repository/org/junit/jupiter/junit-jupiter-api/5.8.2/junit-jupiter-api-5.8.2.jar:/var/lib/jenkins/.m2/repository/company/group/id/company-lib/1.4.5/company-lib-1.4.5.jar::/var/lib/jenkins/.m2/repository/com/fasterxml/jackson/core/jackson-databind/2.10.1/jackson-databind-2.10.1.jar:/var/lib/jenkins/.m2/repository/com/fasterxml/jackson/core/jackson-annotations/2.10.1/jackson-annotations-2.10.1.jar:/var/lib/jenkins/.m2/repository/com/fasterxml/jackson/datatype/jackson-datatype-jdk8/2.10.1/jackson-datatype-jdk8-2.10.1.jar:/var/lib/jenkins/.m2/repository/org/apache/directory/studio/org.apache.commons.lang/2.6/org.apache.commons.lang-2.6.jar:/var/lib/jenkins/.m2/repository/commons-lang/commons-lang/2.6/commons-lang-2.6.jar:/var/lib/jenkins/.m2/repository/com/fasterxml/jackson/core/jackson-core/2.10.1/jackson-core-2.10.1.jar -d /var/lib/jenkins/.jbang/cache/jars/myJangScript.java.8ea89252cad7fa34d37240845c7ff863a7fb53cecaf1b6b4b3bd556dc8eb1d56.jar.tmp /var/lib/jenkins/.jbang/cache/scripts/8ea89252cad7fa34d37240845c7ff863a7fb53cecaf1b6b4b3bd556dc8eb1d56/myJangScript.java
 Picked up JAVA_TOOL_OPTIONS: -Dfile.encoding=UTF-8
 [jbang] Deleting folder /var/lib/jenkins/.jbang/cache/jars/myJangScript.java.8ea89252cad7fa34d37240845c7ff863a7fb53cecaf1b6b4b3bd556dc8eb1d56.jar.tmp
 [jbang] Resolving dependencies...
 [jbang] company.group.id:company-lib:jar:RELEASE
 [jbang] [ERROR] Could not resolve dependencies from mavencentral=https://repo1.maven.org/maven2/
 Unable to collect/resolve dependency tree for a resolution due to: Failed to collect dependencies at company.group.id:company-lib:jar:RELEASE, caused by: Could not find metadata company.group.id:company-lib/maven-metadata.xml in local (/var/lib/jenkins/.m2/repository)
   Failed to collect dependencies at company.group.id:company-lib:jar:RELEASE
 dev.jbang.cli.ExitException: Could not resolve dependencies from mavencentral=https://repo1.maven.org/maven2/
 Unable to collect/resolve dependency tree for a resolution due to: Failed to collect dependencies at company.group.id:company-lib:jar:RELEASE, caused by: Could not find metadata company.group.id:company-lib/maven-metadata.xml in local (/var/lib/jenkins/.m2/repository)
   Failed to collect dependencies at company.group.id:company-lib:jar:RELEASE
 	at dev.jbang.dependencies.DependencyUtil.resolveDependenciesViaAether(DependencyUtil.java:254)
 	at dev.jbang.dependencies.DependencyUtil.resolveDependencies(DependencyUtil.java:122)
 	at dev.jbang.dependencies.DependencyUtil.resolveDependencies(DependencyUtil.java:71)
 	at dev.jbang.dependencies.DependencyResolver.resolve(DependencyResolver.java:67)
 	at dev.jbang.source.RunContext.resolveClassPath(RunContext.java:354)
 	at dev.jbang.source.generators.JarCmdGenerator.generateCommandLineList(JarCmdGenerator.java:48)
 	at dev.jbang.source.generators.BaseCmdGenerator.generate(BaseCmdGenerator.java:35)
 	at dev.jbang.cli.Run.doCall(Run.java:101)
 	at dev.jbang.cli.BaseCommand.call(BaseCommand.java:93)
 	at dev.jbang.cli.BaseCommand.call(BaseCommand.java:15)
 	at picocli.CommandLine.executeUserObject(CommandLine.java:1953)
 	at picocli.CommandLine.access$1300(CommandLine.java:145)
 	at picocli.CommandLine$RunLast.executeUserObjectOfLastSubcommandWithSameParent(CommandLine.java:2352)
 	at picocli.CommandLine$RunLast.handle(CommandLine.java:2346)
 	at dev.jbang.cli.JBang$3.handle(JBang.java:145)
 	at dev.jbang.cli.JBang$3.handle(JBang.java:140)
 	at picocli.CommandLine$AbstractParseResultHandler.execute(CommandLine.java:2179)
 	at picocli.CommandLine.execute(CommandLine.java:2078)
 	at dev.jbang.Main.main(Main.java:14)
 Caused by: org.jboss.shrinkwrap.resolver.api.NoResolvedResultException: Unable to collect/resolve dependency tree for a resolution due to: Failed to collect dependencies at company.group.id:company-lib:jar:RELEASE, caused by: Could not find metadata company.group.id:company-lib/maven-metadata.xml in local (/var/lib/jenkins/.m2/repository)
 	at org.jboss.shrinkwrap.resolver.impl.maven.MavenWorkingSessionImpl.wrapException(MavenWorkingSessionImpl.java:503)
 	at org.jboss.shrinkwrap.resolver.impl.maven.MavenWorkingSessionImpl.resolveDependencies(MavenWorkingSessionImpl.java:242)
 	at org.jboss.shrinkwrap.resolver.impl.maven.MavenStrategyStageBaseImpl.using(MavenStrategyStageBaseImpl.java:70)
 	at org.jboss.shrinkwrap.resolver.impl.maven.MavenStrategyStageBaseImpl.withTransitivity(MavenStrategyStageBaseImpl.java:52)
 	at org.jboss.shrinkwrap.resolver.impl.maven.MavenStrategyStageBaseImpl.withTransitivity(MavenStrategyStageBaseImpl.java:39)
 	at dev.jbang.dependencies.DependencyUtil.resolveDependenciesViaAether(DependencyUtil.java:232)
 	... 18 more
 Caused by: org.eclipse.aether.resolution.DependencyResolutionException: Failed to collect dependencies at company.group.id:company-lib:jar:RELEASE
 	at org.eclipse.aether.internal.impl.DefaultRepositorySystem.resolveDependencies(DefaultRepositorySystem.java:353)
 	at org.jboss.shrinkwrap.resolver.impl.maven.bootstrap.MavenRepositorySystem.resolveDependencies(MavenRepositorySystem.java:121)
 	at org.jboss.shrinkwrap.resolver.impl.maven.MavenWorkingSessionImpl.resolveDependencies(MavenWorkingSessionImpl.java:239)
 	... 22 more
 Caused by: org.eclipse.aether.collection.DependencyCollectionException: Failed to collect dependencies at company.group.id:company-lib:jar:RELEASE
 	at org.eclipse.aether.internal.impl.collect.DefaultDependencyCollector.collectDependencies(DefaultDependencyCollector.java:291)
 	at org.eclipse.aether.internal.impl.DefaultRepositorySystem.resolveDependencies(DefaultRepositorySystem.java:309)
 	... 24 more
 Caused by: org.eclipse.aether.resolution.ArtifactDescriptorException: Failed to read artifact descriptor for company.group.id:company-lib:jar:RELEASE
 	at org.apache.maven.repository.internal.DefaultArtifactDescriptorReader.loadPom(DefaultArtifactDescriptorReader.java:218)
 	at org.apache.maven.repository.internal.DefaultArtifactDescriptorReader.readArtifactDescriptor(DefaultArtifactDescriptorReader.java:171)
 	at org.eclipse.aether.internal.impl.collect.DefaultDependencyCollector.resolveCachedArtifactDescriptor(DefaultDependencyCollector.java:541)
 	at org.eclipse.aether.internal.impl.collect.DefaultDependencyCollector.getArtifactDescriptorResult(DefaultDependencyCollector.java:524)
 	at org.eclipse.aether.internal.impl.collect.DefaultDependencyCollector.processDependency(DefaultDependencyCollector.java:412)
 	at org.eclipse.aether.internal.impl.collect.DefaultDependencyCollector.processDependency(DefaultDependencyCollector.java:365)
 	at org.eclipse.aether.internal.impl.collect.DefaultDependencyCollector.process(DefaultDependencyCollector.java:352)
 	at org.eclipse.aether.internal.impl.collect.DefaultDependencyCollector.collectDependencies(DefaultDependencyCollector.java:254)
 	... 25 more
 Caused by: org.eclipse.aether.resolution.VersionResolutionException: Failed to resolve version for company.group.id:company-lib:jar:RELEASE: Could not find metadata company.group.id:company-lib/maven-metadata.xml in local (/var/lib/jenkins/.m2/repository)
 	at org.apache.maven.repository.internal.DefaultVersionResolver.resolveVersion(DefaultVersionResolver.java:276)
 	at org.apache.maven.repository.internal.DefaultArtifactDescriptorReader.loadPom(DefaultArtifactDescriptorReader.java:204)
 	... 32 more
 Caused by: org.eclipse.aether.transfer.MetadataNotFoundException: Could not find metadata company.group.id:company-lib/maven-metadata.xml in local (/var/lib/jenkins/.m2/repository)
 	at org.eclipse.aether.internal.impl.DefaultMetadataResolver.resolve(DefaultMetadataResolver.java:220)
 	at org.eclipse.aether.internal.impl.DefaultMetadataResolver.resolveMetadata(DefaultMetadataResolver.java:181)
 	at org.apache.maven.repository.internal.DefaultVersionResolver.resolveVersion(DefaultVersionResolver.java:213)
 	... 33 more
 [jbang] If you believe this a bug in jbang, open an issue at https://github.com/jbangdev/jbang/issues
```

Why does JBang compile the script (the classpath looks ok) and then does "Deleting folder" and then "Resolving dependencies..." again…

I hope I will be able to reproduce it locally to understand what is going on and wrong.

This is driving me crazy.

I got once:

```
jbang run --verbose --deps org.example:dummy:1 myJangScript
[jbang] jbang version 0.91.0
[jbang] System Java version detected as 8
[jbang] No build required. Reusing jar from /Users/jbr/.jbang/cache/jars/runSchemaTests.java.0c6512c7977dd97a00fa777d1e360378d0b784da9b8188f0e0b722d8b069c4b1.jar
[jbang] Resolving dependencies...
[jbang] org.example:dummy:jar:1
[jbang] [ERROR] Could not resolve dependencies from mavencentral=https://repo1.maven.org/maven2/
Unable to collect/resolve dependency tree for a resolution due to: Could not find artifact org.example:dummy:jar:1 in mavencentral (https://repo1.maven.org/maven2/), caused by: Could not find artifact org.example:dummy:jar:1 in mavencentral (https://repo1.maven.org/maven2/)
  Could not find artifact org.example:dummy:jar:1 in mavencentral (https://repo1.maven.org/maven2/)
dev.jbang.cli.ExitException: Could not resolve dependencies from mavencentral=https://repo1.maven.org/maven2/
Unable to collect/resolve dependency tree for a resolution due to: Could not find artifact org.example:dummy:jar:1 in mavencentral (https://repo1.maven.org/maven2/), caused by: Could not find artifact org.example:dummy:jar:1 in mavencentral (https://repo1.maven.org/maven2/)
  Could not find artifact org.example:dummy:jar:1 in mavencentral (https://repo1.maven.org/maven2/)
	at dev.jbang.dependencies.DependencyUtil.resolveDependenciesViaAether(DependencyUtil.java:254)
	at dev.jbang.dependencies.DependencyUtil.resolveDependencies(DependencyUtil.java:122)
	at dev.jbang.dependencies.DependencyUtil.resolveDependencies(DependencyUtil.java:71)
	at dev.jbang.dependencies.DependencyResolver.resolve(DependencyResolver.java:67)
	at dev.jbang.source.RunContext.resolveClassPath(RunContext.java:354)
	at dev.jbang.source.generators.JarCmdGenerator.generateCommandLineList(JarCmdGenerator.java:48)
	at dev.jbang.source.generators.BaseCmdGenerator.generate(BaseCmdGenerator.java:35)
	at dev.jbang.cli.Run.doCall(Run.java:101)
	at dev.jbang.cli.BaseCommand.call(BaseCommand.java:93)
	at dev.jbang.cli.BaseCommand.call(BaseCommand.java:15)
	at picocli.CommandLine.executeUserObject(CommandLine.java:1953)
	at picocli.CommandLine.access$1300(CommandLine.java:145)
	at picocli.CommandLine$RunLast.executeUserObjectOfLastSubcommandWithSameParent(CommandLine.java:2352)
	at picocli.CommandLine$RunLast.handle(CommandLine.java:2346)
	at dev.jbang.cli.JBang$3.handle(JBang.java:145)
	at dev.jbang.cli.JBang$3.handle(JBang.java:140)
	at picocli.CommandLine$AbstractParseResultHandler.execute(CommandLine.java:2179)
	at picocli.CommandLine.execute(CommandLine.java:2078)
	at dev.jbang.Main.main(Main.java:14)
Caused by: org.jboss.shrinkwrap.resolver.api.NoResolvedResultException: Unable to collect/resolve dependency tree for a resolution due to: Could not find artifact org.example:dummy:jar:1 in mavencentral (https://repo1.maven.org/maven2/), caused by: Could not find artifact org.example:dummy:jar:1 in mavencentral (https://repo1.maven.org/maven2/)
	at org.jboss.shrinkwrap.resolver.impl.maven.MavenWorkingSessionImpl.wrapException(MavenWorkingSessionImpl.java:503)
	at org.jboss.shrinkwrap.resolver.impl.maven.MavenWorkingSessionImpl.resolveDependencies(MavenWorkingSessionImpl.java:242)
	at org.jboss.shrinkwrap.resolver.impl.maven.MavenStrategyStageBaseImpl.using(MavenStrategyStageBaseImpl.java:70)
	at org.jboss.shrinkwrap.resolver.impl.maven.MavenStrategyStageBaseImpl.withTransitivity(MavenStrategyStageBaseImpl.java:52)
	at org.jboss.shrinkwrap.resolver.impl.maven.MavenStrategyStageBaseImpl.withTransitivity(MavenStrategyStageBaseImpl.java:39)
	at dev.jbang.dependencies.DependencyUtil.resolveDependenciesViaAether(DependencyUtil.java:232)
	... 18 more
Caused by: org.eclipse.aether.resolution.DependencyResolutionException: Could not find artifact org.example:dummy:jar:1 in mavencentral (https://repo1.maven.org/maven2/)
	at org.eclipse.aether.internal.impl.DefaultRepositorySystem.resolveDependencies(DefaultRepositorySystem.java:357)
	at org.jboss.shrinkwrap.resolver.impl.maven.bootstrap.MavenRepositorySystem.resolveDependencies(MavenRepositorySystem.java:121)
	at org.jboss.shrinkwrap.resolver.impl.maven.MavenWorkingSessionImpl.resolveDependencies(MavenWorkingSessionImpl.java:239)
	... 22 more
Caused by: org.eclipse.aether.resolution.ArtifactResolutionException: Could not find artifact org.example:dummy:jar:1 in mavencentral (https://repo1.maven.org/maven2/)
	at org.eclipse.aether.internal.impl.DefaultArtifactResolver.resolve(DefaultArtifactResolver.java:424)
	at org.eclipse.aether.internal.impl.DefaultArtifactResolver.resolveArtifacts(DefaultArtifactResolver.java:229)
	at org.eclipse.aether.internal.impl.DefaultRepositorySystem.resolveDependencies(DefaultRepositorySystem.java:340)
	... 24 more
Caused by: org.eclipse.aether.transfer.ArtifactNotFoundException: Could not find artifact org.example:dummy:jar:1 in mavencentral (https://repo1.maven.org/maven2/)
	at org.eclipse.aether.connector.basic.ArtifactTransportListener.transferFailed(ArtifactTransportListener.java:48)
	at org.eclipse.aether.connector.basic.BasicRepositoryConnector$TaskRunner.run(BasicRepositoryConnector.java:369)
	at org.eclipse.aether.util.concurrency.RunnableErrorForwarder$1.run(RunnableErrorForwarder.java:75)
	at org.eclipse.aether.connector.basic.BasicRepositoryConnector$DirectExecutor.execute(BasicRepositoryConnector.java:644)
	at org.eclipse.aether.connector.basic.BasicRepositoryConnector.get(BasicRepositoryConnector.java:262)
	at org.eclipse.aether.internal.impl.DefaultArtifactResolver.performDownloads(DefaultArtifactResolver.java:499)
	at org.eclipse.aether.internal.impl.DefaultArtifactResolver.resolve(DefaultArtifactResolver.java:401)
	... 26 more
 ```
 
And then after cleaning my local maven repo (JBANG_REPO) and my jbang cache `jbang cache clear` I now have only the error with `acme` (which is good).

--- a/src/main/java/dev/jbang/source/Jar.java
+++ b/src/main/java/dev/jbang/source/Jar.java
@@ -85,6 +85,16 @@ public class Jar implements Code {
 	public SourceSet asSourceSet() {
 		if (sourceSet == null) {
 			sourceSet = SourceSet.forSource(Source.forResourceRef(resourceRef, null));
+			sourceSet.setMainClass(getMainClass());
+			sourceSet.addRuntimeOptions(getRuntimeOptions());
+			sourceSet.setJavaVersion(getJavaVersion().orElse(null));
+			// TODO deduplicate with code from updateDependencyResolver()
+			if (resourceRef.getOriginalResource() != null
+					&& DependencyUtil.looksLikeAGav(resourceRef.getOriginalResource())) {
+				sourceSet.addDependency(resourceRef.getOriginalResource());
+			} else if (classPath != null) {
+				sourceSet.addClassPaths(Arrays.asList(classPath.split(" ")));
+			}
 		}
 		return sourceSet;
 	}

--- a/src/main/java/dev/jbang/source/RunContext.java
+++ b/src/main/java/dev/jbang/source/RunContext.java
@@ -350,7 +350,7 @@ public class RunContext {
 			if (code instanceof Jar) {
 				updateDependencyResolver(resolver);
 			}
-			code.updateDependencyResolver(resolver);
+			code.asSourceSet().updateDependencyResolver(resolver);
 			mcp = resolver.resolve();
 		}
 		return mcp.getClassPath();

--- a/src/main/java/dev/jbang/source/SourceSet.java
+++ b/src/main/java/dev/jbang/source/SourceSet.java
@@ -32,6 +32,7 @@ public class SourceSet implements Code {
 	private String javaVersion;
 	private String description;
 	private String gav;
+	private String mainClass;
 
 	private ModularClassPath mcp;
 	private File jarFile;
@@ -253,6 +254,15 @@ public class SourceSet implements Code {
 	public SourceSet setGav(String gav) {
 		this.gav = gav;
 		return this;
+	}
+
+	@Override
+	public String getMainClass() {
+		return mainClass;
+	}
+
+	public void setMainClass(String mainClass) {
+		this.mainClass = mainClass;
 	}
 
 	@Override

--- a/src/main/java/dev/jbang/source/builders/BaseBuilder.java
+++ b/src/main/java/dev/jbang/source/builders/BaseBuilder.java
@@ -301,15 +301,6 @@ public abstract class BaseBuilder implements Builder {
 		}
 	}
 
-	static public String findMainClass(Path base, Path classfile) {
-		StringBuilder mainClass = new StringBuilder(classfile.getFileName().toString().replace(".class", ""));
-		while (!classfile.getParent().equals(base)) {
-			classfile = classfile.getParent();
-			mainClass.insert(0, classfile.getFileName().toString() + ".");
-		}
-		return mainClass.toString();
-	}
-
 	private static String resolveInGraalVMHome(String cmd, String requestedVersion) {
 		String newcmd = resolveInEnv("GRAALVM_HOME", cmd);
 
@@ -434,7 +425,7 @@ public abstract class BaseBuilder implements Builder {
 				}
 
 				if (!mains.isEmpty()) {
-					ctx.setMainClass(mains.get(0).name().toString());
+					ss.setMainClass(mains.get(0).name().toString());
 					if (mains.size() > 1) {
 						Util.warnMsg(
 								"Could not locate unique main() method. Use -m to specify explicit main method. Falling back to use first found: "

--- a/src/test/java/dev/jbang/cli/TestRun.java
+++ b/src/test/java/dev/jbang/cli/TestRun.java
@@ -645,21 +645,6 @@ public class TestRun extends BaseTest {
 	}
 
 	@Test
-	void testFindMain(@TempDir Path dir) throws IOException {
-
-		File basedir = dir.resolve("a/b/c").toFile();
-		boolean mkdirs = basedir.mkdirs();
-		assert (mkdirs);
-		File classfile = new File(basedir, "mymain.class");
-		classfile.setLastModified(System.currentTimeMillis());
-		classfile.createNewFile();
-		assert (classfile.exists());
-
-		assertEquals(BaseBuilder.findMainClass(dir, classfile.toPath()), "a.b.c.mymain");
-
-	}
-
-	@Test
 	void testCreateJar(@TempDir Path rootdir) throws IOException {
 
 		File dir = new File(rootdir.toFile(), "content");
@@ -1359,7 +1344,7 @@ public class TestRun extends BaseTest {
 
 		ss.builder(ctx).build();
 
-		assertThat(ctx.getMainClass(), equalTo("Two"));
+		assertThat(ss.getMainClass(), equalTo("Two"));
 
 	}
 
@@ -1388,7 +1373,7 @@ public class TestRun extends BaseTest {
 
 		ss.builder(ctx).build();
 
-		assertThat(ctx.getMainClass(), equalTo("One"));
+		assertThat(ss.getMainClass(), equalTo("One"));
 
 	}
 
@@ -1848,6 +1833,36 @@ public class TestRun extends BaseTest {
 			ex.printStackTrace(new PrintWriter(sw));
 			assertThat(sw.toString(), containsString(
 					"Script or alias could not be found or read: 'missing.jsh'"));
+		}
+	}
+
+	@Test
+	void testBuildTwiceWithCliDeps(@TempDir Path output) throws IOException {
+		String script = "" +
+				"//REPOS acme=https://repo1.maven.org/maven2/\n" +
+				"public class script { public static void main(String... args) {} }";
+		JBang jbang = new JBang();
+		Path p = output.resolve("script.java");
+		writeString(p, script);
+
+		CommandLine.ParseResult pr = JBang.getCommandLine().parseArgs("build", p.toFile().getAbsolutePath());
+		Build build = (Build) pr.subcommand().commandSpec().userObject();
+		build.call();
+
+		pr = JBang.getCommandLine().parseArgs("run", "--deps", "org.example:dummy:1", p.toFile().getAbsolutePath());
+		Run run = (Run) pr.subcommand().commandSpec().userObject();
+
+		RunContext ctx = run.getRunContext();
+		Code code = ctx.forResource(p.toString());
+
+		code = run.prepareArtifacts(code, ctx);
+		try {
+			new JarCmdGenerator(code, ctx).generate();
+		} catch (ExitException e) {
+			StringWriter sw = new StringWriter();
+			e.printStackTrace(new PrintWriter(sw));
+			assertThat(sw.toString(), containsString("acme="));
+			assertThat(sw.toString(), not(containsString("mavencentral=")));
 		}
 	}
 }

